src/model/input.ts

```diff
@@ -8,129 +8,195 @@ const core = require('@actions/core');
  * Note that input is always passed as a string, even booleans.
  */
 class Input {
+  public static githubEnabled = true;
+  public static cliOptions;
+  static get cloudRunnerTests(): boolean {
+    return Input.getInput(`CloudRunnerTests`) || false;
+  }
+  private static getInput(query) {
+    return Input.githubEnabled
+      ? core.getInput(query)
+      : Input.cliOptions !== undefined && Input.cliOptions[query] !== undefined
+      ? Input.cliOptions[query]
+      : process.env[query] !== undefined
+      ? process.env[query]
+      : false;
+  }
+  static get region(): string {
+    return Input.getInput('region') || 'eu-west-2';
+  }
+  static get githubRepo(): string {
+    return Input.getInput('GITHUB_REPOSITORY') || 'game-ci/unity-builder';
+    // TODO system get repo remote?
+  }
+  static get branch() {
+    if (Input.getInput(`REMOTE_BUILDER_CACHE`)) {
+      return Input.getInput(`REMOTE_BUILDER_CACHE`);
+    } else if (Input.getInput(`GITHUB_REF`)) {
+      return Input.getInput(`GITHUB_REF`)
+        .split('/')
+        .filter((x) => {
+          x = x[0].toUpperCase() + x.slice(1);
+          return x;
+        })
+        .join('');
+    } else if (Input.getInput('branch')) {
+      return Input.getInput('branch');
+    } else {
+      return 'remote-builder/unified-providers';
+    }
+    // TODO git get branch?
+  }
+
+  static get gitSha() {
+    if (Input.getInput(`GITHUB_SHA`)) {
+      return Input.getInput(`GITHUB_SHA`);
+    }
+    if (Input.getInput(`GitSHA`)) {
+      return Input.getInput(`GitSHA`);
+    }
+    // TODO  git get sha?
```

Comment:
Leftover todo or intended?

Reply:
Will implement

Reply:
This comment refers to the behavior of getting the specific sha to build from git cli. This may make it easy to avoid needing to specify the sha in some cli builds